mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.kt

```diff
@@ -704,36 +682,41 @@ class MainActivity : AbstractBaseActivity(), ConnectionFactory.UpdateListener {
         return wrapped
     }
 
-    private fun openNotificationsPageIfNeeded() {
-        if (pendingOpenedNotificationId != null &&
-            isStarted &&
-            ConnectionFactory.cloudConnection != null
-        ) {
-            openNotifications(pendingOpenedNotificationId)
-            pendingOpenedNotificationId = null
+    private fun executeOrStoreAction(action: PendingAction) {
+        if (!executeActionIfPossible(action)) {
+            pendingAction = action
         }
     }
 
-    private fun openHabpanelIfNeeded() {
-        if (isStarted && shouldOpenHabpanel && serverProperties != null && serverProperties!!.hasHabpanelInstalled()) {
-            openHabpanel()
-            shouldOpenHabpanel = false
+    private fun handlePendingAction() {
+        val action = pendingAction
+        if (action != null && executeActionIfPossible(action)) {
+            pendingAction = null
         }
     }
 
-    private fun launchVoiceRecognitionIfNeeded() {
-        if (isStarted && shouldLaunchVoiceRecognition && serverProperties != null) {
+    private fun executeActionIfPossible(action: PendingAction): Boolean = when {
+        action is PendingAction.ChooseSitemap && isStarted -> {
+            chooseSitemap()
+            true
+        }
+        action is PendingAction.OpenSitemapUrl && isStarted && serverProperties != null -> {
+            buildUrlAndOpenSitemap(action.url)
+            true
+        }
+        action is PendingAction.OpenHabPanel && isStarted && serverProperties?.hasHabpanelInstalled() == true -> {
```

Comment:
`OpenHabPanel` vs `hasHabpanelInstalled`. Can you rename latter to `hasHabPanelInstalled`?